app/models/sales_log.rb

```diff
@@ -348,12 +364,48 @@ def beds_for_la_sale_range
 
   def ownership_scheme
     case ownershipsch
-    when 1
-      "shared ownership"
-    when 2
-      "discounted ownership"
-    when 3
-      "outright sale"
+    when 1 then "shared ownership"
+    when 2 then "discounted ownership"
+    when 3 then "outright sale"
+    end
+  end
+
+private
+
+  def reset_invalidated_dependent_fields!
+    super
+
+    reset_derived_questions!
+  end
+
+  def reset_derived_questions!
+    dependencies = [
+      {
+        parent_conditions: [
+          { attribute: :buylivein, value: 2 },
+        ],
+        derived_attributes: %i[buy1livein buy2livein],
+      },
+      {
+        parent_conditions: [
+          { attribute: :buylivein, value: 1 },
+          { attribute: :jointpur, value: 1 },
+        ],
+        derived_attributes: [:buy1livein],
+      },
+    ]
+
+    dependencies.each do |dependency|
+      any_parent_attributes_changed = dependency[:parent_conditions].any? { |parent_condition| send("#{parent_condition[:attribute]}_changed?") }
+      next unless any_parent_attributes_changed
+
+      were_in_derived_state = dependency[:parent_conditions].all? { |parent_condition| send("#{parent_condition[:attribute]}_was") == parent_condition[:value] }
+      next unless were_in_derived_state
+
+      dependency[:derived_attributes].each do |derived_attribute|
+        Rails.logger.debug("Cleared derived #{derived_attribute} value")
+        send("#{derived_attribute}=", nil)
+      end
```

Comment:
This is worth thinking about and discussing.
I have put comments next to the corresponding method in lettings log so you can see it above.
You will see that I have done it slightly differently, which in itself is undesirable, so for consistency, I propose that either
- we decide the old way was better and I change this to reflect that
- we decide this is better and we mark on the lettings log method to update it to reflect this model during the ticket to clear variables derived from renewal coming up in the next sprint.

this method:
defines the state in which variables are defined, and associated attributes whose values are derived.
for each of this dependencies, the value of the derived attribute is cleared iff the form has started in the state where the derivation occurs && one of the attributes on which the derived attribute depends has changed.
eg the first dependency above says if "Will the buyers live in the property?" is answered "No" (buylivein: 2), then the values for buy1livein and buy2livein are derived, so if we are transitioning away from that parent state, the child attributes should be cleared.

the old method:
defines the state the form would be moving to in order for the derived variable to be cleared. It does not take advantage of the `attribute_was` methods to check if we are transitioning _from_ a state where the derivation would have taken place.
I have put a more concrete example next to the above method or why this is less good.

Reply:
Makes sense to me, I think your justification for the new method is sound. To be clear, are there definite examples where this old/new method distinction will matter?

Reply:
there are not currently situations where this improvement is necessary, however since I was writing this method new I thought it made sense to write it as well as possible.
I think this way is truer to the reality which does give it better future flexibility, but also for me makes it slightly more readable since it better reflects the conditions under which we want to clear things.

Reply:
@jamesrose 
if Nat is happy with this way are you happy that I put a to-do comment on the corresponding method on lettings_log that it should be updated in line with this method as part of ticket 2288 and make a note of this also on the JIRA ticket for 2288?
I ask because this was something we did occasionally on a previous project but I've not seen it on this one.